save-demo/src/main/kotlin/com/saveourtool/save/demo/runners/cli/DiktatCliRunner.kt

```diff
@@ -0,0 +1,137 @@
+package com.saveourtool.save.demo.runners.cli
+
+import com.saveourtool.save.demo.diktat.DiktatDemoAdditionalParams
+import com.saveourtool.save.demo.diktat.DiktatDemoMode
+import com.saveourtool.save.demo.diktat.DiktatDemoResult
+import com.saveourtool.save.demo.diktat.DiktatDemoTool
+import com.saveourtool.save.demo.storage.ToolKey
+import com.saveourtool.save.demo.storage.ToolStorage
+import com.saveourtool.save.demo.utils.prependPath
+import com.saveourtool.save.utils.getLogger
+import com.saveourtool.save.utils.writeToFile
+import io.ktor.util.*
+import org.springframework.stereotype.Component
+import java.io.File
+import java.io.FileNotFoundException
+import java.nio.file.Path
+import kotlin.io.path.*
+
+/**
+ * Class that allows to run diktat/ktlint as command line application
+ *
+ * @property toolStorage
+ */
+@Component
+class DiktatCliRunner(
+    private val toolStorage: ToolStorage,
+) : CliRunner<DiktatDemoAdditionalParams, DiktatDemoResult> {
+    override fun getExecutable(workingDir: Path, params: DiktatDemoAdditionalParams): File {
+        val toolName = params.tool.name.lowercase()
+        val version = if (params.tool == DiktatDemoTool.DIKTAT) {
+            DIKTAT_VERSION
+        } else {
+            KTLINT_VERSION
+        }
+        val fileName = buildString {
+            append(toolName)
+            if (params.tool == DiktatDemoTool.KTLINT) {
+                append("-cli")
+            }
+            if (!osName().startsWith("Linux", ignoreCase = true) && !osName().startsWith("Mac OS", ignoreCase = true)) {
+                append(".cmd")
```

Comment:
This is unnecessary on Windows: you can safely run `diktat` or `diktat.bat` or `diktat.com` or `diktat.exe` as long as the extension is in the `PATHEXT` list:

```console
C:\Users\a00664975>set PATHEXT
PATHEXT=.COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC;.PY;.PYW;.RB;.RBW
```

What you may actually need to do on Windows is prepending the batch script name with `cmd.exe /C` (unlike `.exe` files, bat/cmd scripts are unlikely to be executable &mdash; I doubt they are).

You'll need to quote the whole command line and pass it as a single argument, i.e.

```console
cmd.exe /C "cls && echo Test && pause >nul"
```

will become an array of length 3:

1. `cmd.exe`
2. `/C`
3. `cls && echo Test && pause >nul`

Reply:
Got it, thanks for good guidelines!